tests/test_db_runtime_security.py

```
@@ -683,8 +684,11 @@ def _close_manager() -> None:
         result1, result2 = asyncio.run(run_test())
         self.assertEqual((result1, result2), ("first", "second"))
 
-        with sqlite3.connect(self.db_path) as conn:
-            count = conn.execute("SELECT COUNT(*) FROM test_async_close").fetchone()[0]
+        with contextlib.closing(sqlite3.connect(self.db_path)) as conn:
+            with conn as managed_conn:
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)